mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -550,12 +601,23 @@ def GPU_LaunchFuncOp :GPU_Op<"launch_func", [
     /// The name of the kernel.
     StringAttr getKernelName();
 
+    /// Has cluster
+    bool hasClusterSize() {
+      auto totalSize = getOperands().size();
+      totalSize -= getKernelOperands().size();
+      totalSize -= getAsyncDependencies().size();
+      return totalSize > 7;
```

Comment:
Could we use `getClusterSize() != std::nullopt` instead?

I don't like the hardcoded `7` here.